### DISCUSSION TITLE
当前版本的几个问题

### DISCUSSION THREAD

**Discussion by BANKA2017** [MoeNetwork/Tieba-Cloud-Sign#271] [其他]
本想发到 issue ，但好像也不全是云签本身的问题（？），所以发这边好了

有些是我开发期间自己遇到的，有些是在群聊和贴吧看到的反馈，还有一些是吐槽

## 断签问题

目前常见的意外断签除了帐号真的掉登录以外，还可能是下面两种情况

- 1#用户未登录或登录失败，请更换账号或重试
  实际上 BDUSS 并没有任何问题，重签即可。客户端偶尔把帐号踢下线估计也是这个看起来比较通用的错误导致的（尚未查证）

- 340006#贴吧目录出问题啦，请到贴吧签到吧反馈
  吧主修改了目录导致的，可能出现在新当选吧主操作期间进行的签到（尚未查证）

这两种错误代码都不在重签检查的列表里面，所以会被跳过重签

想要解决也没什么好办法，临时补救办法就是把这两个错误添加到重签的错误代码列表里面，靠重签来兜底

## 贴吧列表数量对不上问题

目前云签使用的移动版网页的方法是我在两年前的[一次提交](https://github.com/MoeNetwork/Tieba-Cloud-Sign/pull/97)引入的，比起最初的扫 i贴吧 列表的方式对系统性能的占用大大降低了，但不好的一方面就是可能会出现列表不完整的问题，甚至会出现关注二十多个贴吧只能扫出个位数的问题，两种方式得到的列表还可能互相有出入。想得到尽可能完整的列表就需要两种方式都扫一遍，但这样消耗就大大地增加了

但，即使这样做也不能保证列表里面的贴吧数目能对上客户端个人中心里面显示的关注数量，我猜消失的是那些被封的贴吧

ps: 在贴吧看到手动添加不上去显示**超出数量限制**的，请站长前往 设置中心->签到设置 找到对应的项目调整数值
ps2: 能不能顺序签到？能，设置位置请看上条

## 签到签不完的问题

- 打乱签到顺序对每天能不能签完影响不大，但在贴吧账号数量比较多的时候能够有效避免打到单个贴吧账号签到过快的限制
- 虽然从来没提过，但实际上每个站点都是有一个能签到的总贴吧数上限的，可以通过 `(24点-开始签到时刻)/计划任务的间隔 * 单表单次签到执行数量 * 分表数量 = 理论贴吧数量上限` 公式计算出不考虑签到失败的理论上限，超出上限的贴吧**一定**无法签到。这个上限对于绝大多数用默认配置私人部署的站点管理员都可以忽略，但那些有十几万甚至几十万贴吧的大站就必须考虑影响了
- `单表单次签到执行数量 * 分表数量` 的值也不可能无限增长，每个签到程序在间隔时间内能发出的请求数量是有限的（因为默认不开并发请求，自带的并发似乎还有点 bug），可能还要考虑 PHP 脚本超时的问题

## 数据库/网络/Nginx/Caddy/Docker/插件 的各种问题

不知道，玄学问题无法复现我也很无奈

## 错误追踪

我同意云签的错误追踪确实约等于没有，但你问我有没有好办法，我只能说没有……有想法请在底下畅所欲言，我就摆张截图：尝试追踪 `msg()` 函数的调用来追踪错误来源，但这样对于通过重定向展示错误的情况仍然是无解的
![image](https://github.com/user-attachments/assets/59002af4-2dd8-4297-99e3-357ed2c5f157)

## 说了那么多，为什么不提 pr，你可是维护者之一欸？

提 pr / commit 的前提是真能解决问题，提交了不保证能解决问题，甚至可能会带来更多的问题，那种干了活还白费功夫的感觉真的不好受

**Comment by n0099** [comment]
https://github.com/BANKA2017/tbsign_go/issues/1
